packages/std/src/ibc.rs

```diff
@@ -63,6 +54,29 @@ pub struct IbcEndpoint {
     pub channel_id: String,
 }
 
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum IbcTimeout {
+    /// block timestamp (nanoseconds since UNIX epoch) after which the packet times out
+    /// (measured on the remote chain)
+    /// See https://golang.org/pkg/time/#Time.UnixNano
+    TimestampNanos(u64),
+    /// block after which the packet times out (measured on remote chain)
+    Block(IbcTimeoutBlock),
+    Both {
+        timestamp_nanos: u64,
+        block: IbcTimeoutBlock,
+    },
+}
+
+impl IbcTimeout {
+    pub fn in_secs(block: &BlockInfo, secs: u64) -> Self {
+        let secs = block.time + secs;
+        let nanos = secs * 1_000_000_000 + block.time_nanos;
+        IbcTimeout::TimestampNanos(nanos)
+    }
```

Comment:
I'm not a fan of this tight coupling between the two types `IbcTimeout` and `BlockInfo`. I don't think `IbcTimeout` should need to know `BlockInfo` exists.

However, I see the point. We need some `now + x` constructor, which is a pain as we do not have a nanosecond precision time point type.

Could you move this logic back to the contract and we come up with a better solution later? Otherwise we risk the need for breaking changes.

Reply:
What is the problem here?
`BlockInfo` is a super basic primitive. This is to avoid the very typical problem (that I hit a few times) of passing in seconds into the timeout field - both u64.

I think this is the simplest and most straight-forward approach to that. I can pull this into a separate function and not a method if you want, but I would like to keep it in `cosmwasm_std`

Reply:
A number of types and contracts depend on `BlockInfo` (like all the `Expiration` times)

Reply:
> What is the problem here?

The problem is we throw all of `BlockInfo` into `IbcTimeout` that does not care about the block info at all (height, chain ID) and simply needs a proper time format. The time in `BlockInfo` is not a great Rust type. It is designed for JSON compatibility.

I can create a better solution that as part of this PR if you want. My idea was to unblock this PR by avoiding standard library design discussions here.

> A number of types and contracts depend on `BlockInfo` (like all the `Expiration` times)

No type in cosmwasm-std depends on `BlockInfo` other than `Env`.

Reply:
Fine, I will move this to the contracts, but I don't see that as more than a placeholder.

Reply:
Ah, I see you made the Timepoint. Thanks